tools/hpcc_aks_util.pl

```diff
@@ -0,0 +1,772 @@
+
+COST CALCULATIONS:
+=================
+
+The script uses East US 2024 pricing as baseline and applies configurable discounts:
+
+--ea-discount           Applied to most VM series (D-series v5, v4 smaller, E-series, F-series, B-series)
+--ds_v4_family_discount Applied specifically to D16ds v4 and larger instances
+
+Default discounts are 10% each. Costs are estimates and actual pricing may vary.
+
+MESSAGES SECTION:
+================
+
+The script provides actionable insights:
+
+- Resource Request Issues: Identifies pods without CPU requests (prevents proper scheduling)
+- Pod Health: Reports pods that are not ready with reasons
+- Thor Conflicts: Detects thor agents and workers/managers in same pool (prevents scaling to 0)
+- Resource Waste: Identifies pools with no target namespace workloads but consuming CPU
+
+UNDERSTANDING THE METRICS:
+=========================
+
+Key efficiency indicators:
+- HPCC(%) shows how much of the pool is used by your target namespace
+- InUse(%) shows actual CPU consumption vs. what's available
+- High Alloc(%) with low InUse(%) may indicate over-provisioning
+- Cost/Hr helps identify expensive pools that might be candidates for optimization
+
+For optimal resource utilization:
+- HPCC(%) should be high for dedicated pools
+- Pods should have appropriate CPU/memory requests
+- No thor conflicts should exist
+- Consider scaling down pools with 0% HPCC but >0% InUse
+
+EOF
+}
+
+# Initialize variables
+my $resource_group = "";
+my $cluster_name = "";
+my $namespace = "";
+my $show_pods = 0;
+my $show_nodes = 0;
+my $ea_discount = 10;
+my $ds_v4_family_discount = 10;
+my $help = 0;
+my $help_detailed = 0;
+
+# Parse command line arguments
+GetOptions(
+    'g|resource-group=s' => \$resource_group,
+    'name|cluster-name=s' => \$cluster_name,
+    'namespace=s' => \$namespace,
+    'show-pods' => \$show_pods,
+    'show-nodes' => \$show_nodes,
+    'ea-discount=i' => \$ea_discount,
+    'ds_v4_family_discount=i' => \$ds_v4_family_discount,
+    'h|help' => \$help,
+    'help-detailed' => \$help_detailed
+) or do {
+    print "Error: Unknown option\n";
+    show_usage();
+    exit 1;
+};
+
+# Show help if requested
+if ($help) {
+    show_usage();
+    exit 0;
+}
+
+# Show detailed help if requested
+if ($help_detailed) {
+    show_detailed_help();
+    exit 0;
+}
+
+# Validate required parameters
+if (!$resource_group || !$cluster_name || !$namespace) {
+    print "Error: Missing required parameters.\n\n";
+    show_usage();
+    exit 1;
+}
+
+# Function to get estimated hourly cost for Azure VM sizes (East US pricing as of 2024)
+# Includes configurable Enterprise Agreement and Dds_v4 family discounts
+sub get_vm_hourly_cost {
+    my ($vm_size) = @_;
+    my $ea_multiplier = (100 - $ea_discount) / 100;
+    my $ds_v4_family_multiplier = (100 - $ds_v4_family_discount) / 100;
+    
+    my %vm_costs = (
+        # Standard D-series v5 (EA discount applied)
+        'Standard_D2s_v5' => 0.096 * $ea_multiplier,
```

Comment:
future: these hard-coded figures should be in a table somewhere, ideally in an external file, so the script doesn't need updaating